pennylane/interfaces/jax.py

```diff
@@ -596,3 +598,8 @@ def _to_jax(res):
                     sub_r.append(jnp.array(r_i))
             res_.append(tuple(sub_r))
     return res_
+
+
+def _to_jax_shot_vector(res):
+    res_ = [tuple(_to_jax([r_])[0] for r_ in r) for r in res]
```

Comment:
Why not pass in `r` directly to `_to_jax`? I.e., `_to_jax(r)` or `_to_jax(list(r))`.

Also, can we be sure here that `r` is a sequence/arraylike?

I think it would be good to have a docstring here to comment on the expected structure of `res` just to improve readability.

Reply:
Doc string added for both. For `_to_jax` you assume that the res are results of multiple tape execution (list). Where for shot vectors case we assume that the shape is a list of results that are tuple of len(shot_vector).